apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts

```diff
@@ -26,20 +26,24 @@ const isValidUrl = (url: string) => {
 };
 
 export const useGrpcEndpointForm = () => {
```

Comment:
This component is pretty complex. When we have more time, we should think of an alternative design/paradigm to manage this state. At the moment, it's react-hook hell.

Reply:
Yes, I have it on my sticky note to do so. I have an inclination that a finite state machine pattern would be better. However, there's always the pesky "reactive" thing that needs to hook into the react lifecycle.